app/lib/common/utils/routes.dart

```diff
@@ -39,6 +39,7 @@ enum Routes {
   spaces('/spaces'),
   joinSpace('/spaces/join'),
   createSpace('/spaces/create'),
+  linkRoom('/:spaceId([!#][^/]+)/linkRoom'),
```

Comment:
I'd still make this three different routes, that also elevates that query param problem....

```suggestion
  linkSubspace('/:spaceId([!#][^/]+)/linkSubspace'),
  linkChat('/:spaceId([!#][^/]+)/linkChat'),
  linkRecommended('/:spaceId([!#][^/]+)/linkRecommended'),
```